lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -50,6 +45,14 @@ class LimitSubqueryOutputWalker extends SqlWalker
 {
     private const ORDER_BY_PATH_EXPRESSION = '/(?<![a-z0-9_])%s\.%s(?![a-z0-9_])/i';
 
+    private const SUPPORTS_ROW_NUMBER = [
+        'db2' => true,
+        'mssql' => true,
+        'oracle' => true,
+        'postgresql' => true,
+        'sqlanywhere' => true,
+    ];
```

Comment:
Let's use a list + `in_array` here too?

Reply:
I wasn't sure if `platformSupportsRowNumber()` might be on a hot execution path. A lookup by array-key is usually faster than a search.

Reply:
I see. I don't think it's a hot path, we're in an output walker, the hot path in Doctrine would rather be in hydration-related features I think.

Reply:
ping @derrabus :slightly_smiling_face:

Reply:
Now that `getName()` is deprecated, I need to solve this entirely different anyway. 🤷

Reply:
Oh right, indeed 👍